lib/util/http/github.ts

```diff
@@ -172,6 +175,66 @@ function constructAcceptString(input?: any): string {
   return acceptStrings.join(', ');
 }
 
+function getGraphqlPageSize(
+  baseUrl: string,
+  fieldName: string,
+  initialCount = 100
+): number {
+  // istanbul ignore if
+  if (ensureTrailingSlash(baseUrl) !== ensureTrailingSlash(githubBaseUrl)) {
+    return initialCount;
+  }
+
+  let count = initialCount;
+
+  const cache = getCache();
+  cache.githubGraphql ??= {};
+  const cachedRecord = cache.githubGraphql[fieldName];
+  delete cache.githubGraphql[fieldName];
+
+  const now = DateTime.local();
+  let timestamp = now.toISO();
+
+  if (cachedRecord) {
+    logger.debug(
+      { fieldName, ...cachedRecord },
+      'Found cached GraphQL page size'
+    );
+    const then = DateTime.fromISO(cachedRecord.timestamp);
+    const expiry = then.plus({ hours: 24 });
+    const isExpired = now > expiry;
+    timestamp = isExpired ? now.toISO() : then.toISO();
+    count = isExpired ? Math.min(count * 2, 100) : cachedRecord.count;
+  }
+
+  if (count < 100) {
+    cache.githubGraphql[fieldName] = { timestamp, count };
```

Comment:
Seems strange to me to set the cache inside a getX function while also setting it in the setX

Reply:
That's because we have to increase page size twice when time is out, and most cleaner way to found it was to erase the whole record and write it again if the resulting page size is different than default one.